### PR TITLE
[Auth] #415 auth instance

### LINKED ISSUE #415: Auth Instance wrapper
follow the same pattern as in Sign

### PR DESCRIPTION
**Pull request by llbartekll**
# Description
Add Auth singletone instance that wraps auth client

Resolves #415 

## How Has This Been Tested?
- needs to be tested in example app [#451](https://github.com/WalletConnect/WalletConnectSwiftV2/issues/451)

## Due Dilligence

* [ ] Breaking change
* [ ] Requires a documentation update

### CODE REVIEW

File: Sources/Auth/Auth.swift [hunk 1]
```diff
@@ -0,0 +1,26 @@
+import Foundation
+import WalletConnectRelay
+import Combine
+
+public class Auth {
+
+    public static var instance: AuthClient = {
+        guard let metadata = Auth.metadata else {
+            fatalError("Error - you must call Sign.configure(_:) before accessing the shared instance.")
```
Comment (flypaper0) [inline Sources/Auth/Auth.swift:9]: Auth.configure(_:)

File: Sources/Auth/Auth.swift [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+            relayClient: Relay.instance)
+    }()
+
+    private static var metadata: AppMetadata?
+    private static var account: Account?
+
+    private init() { }
+
+    static public func configure(metadata: AppMetadata, account: Account?) {
+        Auth.metadata = metadata
```
Comment (flypaper0) [inline Sources/Auth/Auth.swift:23]: mb config if more than 1 property?

File: Sources/Auth/Services/Wallet/WalletRequestSubscriber.swift [hunk 3]
```diff
@@ -19,15 +19,15 @@ class WalletRequestSubscriber {
...
+            subscribeForRequest()
+        }
     }
 
     private func subscribeForRequest() {
-        guard let address = address else {
-            logger.warn("unexpected request")
-            return
-        }
+        guard let address = address else {return}
```
Comment (flypaper0) [inline Sources/Auth/Services/Wallet/WalletRequestSubscriber.swift:28]: 👍

File: Sources/Auth/Services/Wallet/WalletRequestSubscriber.swift [hunk 4]
```diff
@@ -19,15 +19,15 @@ class WalletRequestSubscriber {
         self.logger = logger
         self.address = address
         self.messageFormatter = messageFormatter
-        subscribeForRequest()
+        if address != nil {
```
Comment (flypaper0) [inline Sources/Auth/Services/Wallet/WalletRequestSubscriber.swift:22]: we checking this in subscribeForRequest

File: Sources/Auth/Auth.swift (hunk 2 above)
Comment (llbartekll) [reply Sources/Auth/Auth.swift:23]: relay have more and we do not use config.
you removed the Config and actually I liked it

**Review by flypaper0 — APPROVED** [review]

File: Sources/Auth/Services/Wallet/WalletRequestSubscriber.swift (hunk 4 above)
Comment (llbartekll) [reply Sources/Auth/Services/Wallet/WalletRequestSubscriber.swift:22]: true it is redundant, will remove it

**Pull request merged**